compiler.ts

```diff
@@ -209,6 +211,28 @@ function codeGenExpr(expr : Expr<Type>, env: GlobalEnv) : Array<string> {
       var callName = expr.name;
       if (expr.name === "print" && argTyp === NUM) {
         callName = "print_num";
+      } else if (expr.name === "print" && argTyp === STRING) {
+        callName = "print_str";
```

Comment:
I don't think we should run a loop each time we call this function. Can we just pass the address to print and have `print` responsible for finding these characters on the heap and printing out a single string? I don't think we want thousands of calls to `print` for printing a long string.

Reply:
That would also avoid the need to have these extra `$` variables in the generated code.

You will need to do some research/ask us about how to convert from memory into a JS string from JS, but that is totally doable.

Reply:
Got it. So, is there a way to access the js memory in webstart.ts ?(Can you provide any links to some documentation if you know any) 
You're right, that way I can just send the start address by calling print_str once and it prints out the string.
Thanks!